app/code/community/Sitewards/B2BProfessional/Model/Observer.php

```diff
@@ -251,9 +251,12 @@ protected function removePriceFilter($oBlock)
     {
         $aFilterableAttributes    = $oBlock->getData('_filterable_attributes');
         $aNewFilterableAttributes = array();
-        foreach ($aFilterableAttributes as $oFilterableAttribute) {
-            if ($oFilterableAttribute->getAttributeCode() != 'price') {
-                $aNewFilterableAttributes[] = $oFilterableAttribute;
+        
+        if (is_array($aFilterableAttributes) || is_object($aFilterableAttributes)) {
```

Comment:
So, looking at openmage (https://github.com/OpenMage/magento-mirror/blob/505e8e982ad5f4fd22005060564cad210aab7b6f/app/code/core/Mage/Catalog/Model/Layer.php#L215) this looks it's either going to be a collection of the type `Mage_Catalog_Model_Resource_Eav_Mysql4_Attribute_Collection` or an array, as we're overwriting the collection from earlier below. The Hungarian notation of the variable (the `a` part of the `aFilterableAttributes` should thus be `m`).

@jhoelzl If I understand correctly, you're using the `is_array` and `is_object` checks to determine if something that's iterable exists, rather than attempting to `foreach` over null -- is that correct? I am not super familiar with this code. 

If that is the case, is a check for `is_iterable` perhaps the more expressive check?